src/github.com/matrix-org/dendrite/clientapi/auth/auth.go

```diff
@@ -86,6 +86,18 @@ func GenerateAccessToken() (string, error) {
 	return base64.RawURLEncoding.EncodeToString(b), nil
 }
 
+// GenerateDeviceID creates a new device id. Returns an error if failed to generate
+// random bytes.
+func GenerateDeviceID() (string, error) {
+	b := make([]byte, tokenByteLength)
```

Comment:
let's not have 42-character device IDs. The current ones are 8 characters. Sorry, but you probably need to do this the hard way, with reference to the database.